gax-grpc/src/test/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProviderTest.java

```diff
@@ -157,6 +157,45 @@ public void testWithPoolSize() throws IOException {
     }
   }
 
+  @Test
+  public void testToBuilder() {
+    Duration keepaliveTime = Duration.ofSeconds(1);
+    Duration keepaliveTimeout = Duration.ofSeconds(2);
+    ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder> channelConfigurator =
+        new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+          @Override
+          public ManagedChannelBuilder apply(ManagedChannelBuilder input) {
+            throw new UnsupportedOperationException();
+          }
+        };
+    Map<String, ?> directPathServiceConfig = ImmutableMap.of("loadbalancingConfig", "grpclb");
+
+    InstantiatingGrpcChannelProvider provider =
+        InstantiatingGrpcChannelProvider.newBuilder()
+            .setProcessorCount(2)
+            .setEndpoint("fake.endpoint:443")
+            .setMaxInboundMessageSize(12345678)
+            .setMaxInboundMetadataSize(4096)
+            .setKeepAliveTime(keepaliveTime)
+            .setKeepAliveTimeout(keepaliveTimeout)
+            .setKeepAliveWithoutCalls(true)
+            .setChannelConfigurator(channelConfigurator)
+            .setChannelsPerCpu(2.5)
+            .setDirectPathServiceConfig(directPathServiceConfig)
+            .build();
+
+    InstantiatingGrpcChannelProvider.Builder builder = provider.toBuilder();
+
+    assertThat(builder.getEndpoint()).isEqualTo("fake.endpoint:443");
```

Comment:
gratuitous use of Truth. assertEquals is fully adequate here

Reply:
Did you mean `org.junit.Assert.assertEquals()` as I did't find `Truth. assertEquals()`?  For `org.junit.Assert.assertEquals()` we tend to put expected value and actual value in wrong order and its error message is not as detailed as `Truth. assertEquals()`.